daffodil-core/src/main/scala/org/apache/daffodil/dsom/ModelGroup.scala

```diff
@@ -83,9 +83,6 @@ object ModelGroupFactory {
           val groupRefInstance = groupRefFactory.groupRef
           List(groupRefInstance.asModelGroup)
         }
-        case <annotation>{ _* }</annotation> => Nil
-        case textNode: Text => Nil
-        case _: Comment => Nil
         case _ => {
           lexicalParent.SDE("Unrecognized construct: %s", child)
```

Comment:
I think this should not be an SDE. It should be Assert.invariantFailed("Unrecognized construct %s should be handled by caller.".format(child)), because if the code gets here with an annotation or comment node of XML that's not a user's problem with their schema, but a flaw in our code (likely a mistake during maintenance).

Reply:
Updates made!